Tabulate doc coverage output using markdown

vvvito opened the pull request:
Resolves #46 

## What has changed?

Introduces a new argument called `cov_format` that accepts `string` (default) or `markdown` for the `compute doc` command.

Outputs the results to the terminal in either the default string format, or in a markdown friendly format

### Example

```bash
dbt-coverage compute doc --cov-format markdown
```

![image](https://user-images.githubusercontent.com/87679783/201132423-d8db9413-ca1b-4851-b520-969b15e5391d.png)

results in:

![image](https://user-images.githubusercontent.com/87679783/201132964-cabfeeee-92f9-4918-86ab-372a91f91eac.png)

mrshu commented on dbt_coverage/__init__.py:
```diff
@@ -616,7 +636,7 @@ def fail_compare(coverage_report: CoverageReport, compare_path: Path):
 def do_compute(project_dir: Path = Path('.'), cov_report: Path = Path('coverage.json'),
                cov_type: CoverageType = CoverageType.DOC, cov_fail_under: float = None,
-               cov_fail_compare: Path = None):
+               cov_fail_compare: Path = None, cov_format: Path = Path('string')):
```
@vvvito would you mind using the [`Enum`](https://typer.tiangolo.com/tutorial/parameter-types/enum/) type in here instead? I believe it semantically makes a bit more sense than `Path` and would allow us to do a bit nice comparison on line 649 and at the same time make `cov_format` future proof (in the sense that we could add other formats in the future too 🙂)

Thanks!

vvvito replied:
You got it boss @mrshu.
Updated all references with new enum `CoverageFormat`  on line 29. 
Two entries containing either `STRING_TABLE` or `MARKDOWN_TABLE`. Unsure of the naming convention - can be refactored in the future?

mrshu replied:
Thanks @vvvito, that totally works 🙂

mrshu commented on dbt_coverage/__init__.py:
```diff
@@ -625,8 +645,11 @@ def do_compute(project_dir: Path = Path('.'), cov_report: Path = Path('coverage.
-
-    print(coverage_report.to_formatted_string())
+    
+    if str(cov_format) == 'markdown':
```
@vvvito same as above -- I believe the [`Enum`](https://typer.tiangolo.com/tutorial/parameter-types/enum/) would make this much nicer to handle 🙂

mrshu commented on dbt_coverage/__init__.py:
```diff
@@ -664,10 +687,12 @@ def compute(project_dir: Path = typer.Option('.', help="dbt project directory pa
                                                              "coverage drop between subsequent "
-                                                             "tests.")):
+                                                             "tests."),
+            cov_format: Path = typer.Option('string', help="The output format to print, either "
```
@vvvito just as I mentioned above, I think it would be great to use the [`Enum`](https://typer.tiangolo.com/tutorial/parameter-types/enum/) here instead -- it would ensure the choices are then nicely documented by themselves 🙂

mrshu reviewed (changes requested):
Thank you for the PR @vvvito!

It looks rather nice -- I had only one suggestion about the use of [`Enum`](https://typer.tiangolo.com/tutorial/parameter-types/enum/) -- do let me know if that would make sense to you.

Also, would you mind adding yourself to the `CHANGELOG.md` just as on the link below? That would help us make sure we attribute your work correctly 🙂 

https://github.com/slidoapp/dbt-coverage/pull/38/files#diff-06572a96a58dc510037d5efa622f9bec8519bc1beab13c9f251e97e657a9d4edR9 

Thanks!

vvvito commented:
Thanks @mrshu 
Updated `cov_format` argument to now accept a `CoverageFormat` enum object with either 'string' or 'markdown' as potential values - resolving to `STRING_TABLE` or `MARKDOWN_TABLE` enum entries.

Also updated Changelog with attribution and resolved conflicts between other PRs

mrshu commented on dbt_coverage/__init__.py:
```diff
@@ -756,11 +784,13 @@ def compute(project_dir: Path = typer.Option('.', help="dbt project directory pa
                                                                              "filter tables "
-                                                                             "on.")):
+                                                                             "on."),
+            cov_format: CoverageFormat = typer.Option(None, help="The output format to print, either "
```
@vvvito do you think we could default to `CoverageFormat.STRING_TABLE` here? I guess that would make it a bit more straightforward in the help message to pick up what the default is 🙂

mrshu reviewed (approved):
Thanks a bunch @vvvito, just one last comment and this is good to go!

mrshu commented:
Thanks again @vvvito, this was now released in `0.3.0` (the latest version on PyPI)! 

I also took the liberty of adding a short section on it to the docs: https://github.com/slidoapp/dbt-coverage#markdown-output-with---cov-format

I hope you like it!
